File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        current_min_sell = 1000000
        res = 0
        for price in prices:
            if price-current_min_sell >= res:
                res = price-current_min_sell
            if price < current_min_sell:
                current_min_sell = price
        return res

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int curr_min = 1000000;
        int res = 0;
        for (int i = 0; i < prices.length; i++) {
            int k = prices[i];
            if (k-curr_min > res) {
                res = k-curr_min;
            }
            if (k < curr_min) {
                curr_min = k;
            }
        }
        return res;
    }
}